lib/portlayer/exec/exec.go

```diff
@@ -113,11 +113,14 @@ func Init(ctx context.Context, sess *session.Session, source extraconfig.DataSou
 func eventCallback(ie events.Event) {
 	// grab the container from the cache
 	container := Containers.Container(ie.Reference())
+
 	if container != nil {
+		container.m.Lock()
```

Comment:
Does this re-introduce the deadlock? During `Container.Commit` we hold on to the lock during vsphere operations. If this callback happens during one of those operations, and the operation stretches long), won't we deadlock like before?

Also we have not addressed @caglar10ur original source of the problem: caching the state of the container in the handle.